kingpin/actors/aws/cloudformation.py

```diff
@@ -230,6 +232,33 @@ def _discover_default_params(self, template_body):
         }
         return default_params
 
+    def _strip_hash(self, template_str: str, as_obj: bool = False):
+        """Strips the hash from the template.
+
+        Note: This will also strip the "Outputs" section if no other output exists. This might cause
+        issues when diffiing a template that contains an outputs section with no outputs.
+        """
+
+        template_obj = json.loads(template_str)
+
+        if len(KINGPIN_CFN_HASH_OUTPUT_KEY) > 0:
```

Comment:
1. Why are we doing a length check? Why not:
```python
if KINGPIN_CFN_HASH_OUTPUT_KEY:
  ...
```

2. It's a bit more "pythonic" to do something like:
```python
if !KINGPIN_CFN_HASH_OUTPUT_KEY:
  return template_str
``

Reply:
I will test how python handles env vars and the difference between passing in an empty string vs a "null" (in bash). I thought it would be easier to just keep it as a string and do a length comparison. That way a user could disable the hash-management functionality by passing an empty string where as the default case would be that we do use the hash functionality.